options_test.go

```diff
@@ -1,40 +1,90 @@
 package envbuilder_test
 
 import (
+	"bytes"
 	"testing"
 
 	"github.com/coder/envbuilder"
 	"github.com/coder/serpent"
 	"github.com/stretchr/testify/require"
 )
 
-type testEnvCase struct {
-	Env      string
-	EnvValue string
-	Expected interface{}
-	Actual   func() interface{}
-}
+// TestEnvOptionParsing tests that given environment variables of different types are handled as expected.
+func TestTestEnvOptionParsing(t *testing.T) {
+	t.Run("string", func(t *testing.T) {
+		t.Setenv("SETUP_SCRIPT", "setup.sh")
+		t.Setenv("INIT_COMMAND", "sleep infinity")
+
+		var o envbuilder.Options
+		err := runCLI(&o)
+		require.NoError(t, err)
+
+		require.Equal(t, o.SetupScript, "setup.sh")
+		require.Equal(t, o.InitCommand, "sleep infinity")
+	})
+
+	t.Run("int", func(t *testing.T) {
+		t.Setenv("CACHE_TTL_DAYS", "7")
+
+		var o envbuilder.Options
+		err := runCLI(&o)
+		require.NoError(t, err)
+
+		require.Equal(t, o.CacheTTLDays, int64(7))
+	})
+
+	t.Run("string array", func(t *testing.T) {
+		t.Setenv("IGNORE_PATHS", "/var,/temp")
+
+		var o envbuilder.Options
+		err := runCLI(&o)
+		require.NoError(t, err)
 
-func TestEnvOptions(t *testing.T) {
-	t.Setenv("SETUP_SCRIPT", "setup.sh")
-	t.Setenv("CACHE_TTL_DAYS", "7")
-	t.Setenv("IGNORE_PATHS", "/var,/tmp")
-	t.Setenv("SKIP_REBUILD", "true")
-	t.Setenv("GIT_CLONE_SINGLE_BRANCH", "")
+		require.Equal(t, o.IgnorePaths, []string{"/var", "/temp"})
+	})
 
-	var o = envbuilder.Options{}
+	t.Run("bool", func(t *testing.T) {
+		t.Setenv("SKIP_REBUILD", "true")
+		t.Setenv("GIT_CLONE_SINGLE_BRANCH", "false")
+		t.Setenv("EXIT_ON_BUILD_FAILURE", "true")
+		t.Setenv("FORCE_SAFE", "false")
+		t.Setenv("INSECURE", "true")
```

Comment:
You're currently only testing one sort of coercion: lowecase string->bool. There are others which are supported by https://pkg.go.dev/strconv#ParseBool.

```suggestion
		t.Setenv("SKIP_REBUILD", "TRUE")
		t.Setenv("GIT_CLONE_SINGLE_BRANCH", "FALSE")
		t.Setenv("EXIT_ON_BUILD_FAILURE", "1")
		t.Setenv("FORCE_SAFE", "0")
		t.Setenv("INSECURE", "true")
```

Reply:
What if we don't have enough options to test all of them? 😅

Reply:
Looks like we have enough to test all those cases?

Reply:
it would be missing:
- Empty value as false
- "false" in lowercase as false

Reply:
Ah. You could test the same options below this set of tests, or break them up into sections?

Like:

`true`<br>
`false`

---

`TRUE`<br>
`FALSE`

---

`1`
`0`

---

`<empty>`

---

With things like this it's better to be comprehensive rather than a customer running into strange behaviour which is difficult to track down